### PR TITLE
fix(`MessageService`): fix error when deleting message with reactions

### LINKED ISSUE #302: reaction should be deleted before message

### PR DESCRIPTION
**Pull request by JonasMuehlmann**
# Before this PR can be merged, the following checkboxes have to be ticked

- [x] This code belongs to the backend and is tested, or tests are not useful
- [x] New functions with more than 5 lines have docstrings
- [x] New code has been integrated with the `MessengerService` if appropiate
- [x] New code has been integrated with the `SignalRService` if appropiate
- [x] New functions or classes have been documented in the _Entwurfsdokumentation_
- [x] Changes introduce no commented out code or `Console.Writeline()` statements used for debugging

# Related issues:
Closes  #302

### CODE REVIEW

**Comment by JonasMuehlmann** [comment]
Note:All commits except a6cb4a00eb39e382f4949cca7a7c4e2d9e98deb0 are results of merging main into this branch.

File: Messenger/Messenger.Core/Services/MessageService.cs [hunk 1]
```diff
@@ -192,13 +192,23 @@ public static async Task<bool> DeleteMessage(uint messageId)
...
             var replies = await RetrieveReplies(messageId);
 
             foreach (var reply in replies)
             {
                 await DeleteMessage(reply.Id);
             }
 
+            // Delete reactions to prevent errors
+            var reactions = await RetrieveReactions(messageId);
+
```
Comment (JayKim94) [inline Messenger/Messenger.Core/Services/MessageService.cs:205]: could you add null checks here? foreach would throw error if `reactions` == null

File: Messenger/Messenger.Core/Services/MessageService.cs [hunk 2]
```diff
@@ -192,13 +192,23 @@ public static async Task<bool> DeleteMessage(uint messageId)
             LogContext.PushProperty("SourceContext", "MessageService");
             logger.Information($"Function called with parameters messageId={messageId}");
 
+            // Delete replies to prevent errors
             var replies = await RetrieveReplies(messageId);
```
Comment (JayKim94) [inline Messenger/Messenger.Core/Services/MessageService.cs:197]: could you add null checks here? foreach would throw error if reactions == null

**Review by JayKim94 — CHANGES_REQUESTED** [review]

File: Messenger/Messenger.Core/Services/MessageService.cs (hunk 1 above)
Comment (JonasMuehlmann) [reply Messenger/Messenger.Core/Services/MessageService.cs:208]: Fixed in 07a20b9

File: Messenger/Messenger.Core/Services/MessageService.cs (hunk 2 above)
Comment (JonasMuehlmann) [reply Messenger/Messenger.Core/Services/MessageService.cs:197]: Fixed in 07a20b9

**Review by JayKim94 — APPROVED** [review]

**Pull request merged**